src/main/java/com/example/webapp/Service/DepartmentService.java

```diff
@@ -2,58 +2,65 @@
 
 import com.example.webapp.Model.Department;
 import com.example.webapp.Repository.DepartmentRepository;
+import com.example.webapp.dto.DepartmentDTO;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 
 @Service
 public class DepartmentService {
-@Autowired
-DepartmentRepository departmentRepository;
-    public ResponseEntity<List<Department>> getAll(){
-        try{
-            List<Department> departmentList = new ArrayList<>();
-            departmentRepository.findAll().forEach(departmentList::add);
-            if(departmentList.isEmpty()){
-                return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-            }
-            return new ResponseEntity<>(departmentRepository.findAll(), HttpStatus.OK);
-        } catch (Exception e){
-            return new ResponseEntity<>(null, HttpStatus.INTERNAL_SERVER_ERROR);
+    @Autowired
+    DepartmentRepository departmentRepository;
```

Comment:
Would recommend constructor injection. Also when you have constructor injection this variable can be declared as final

Reply:
Ok